### PR TITLE
feat: Update and `main.py` refactoring

### LINKED ISSUE #127: Update base for `viur-core` `3.6.X`
#128 is the branch for this.

### PR DESCRIPTION
**Pull request by phorward**
- Updates
  - Python 3.12
  - viur-core==3.6.6
  - viur-cli==2.0.1
  - viur-admin==4.1.6
- `main.py`
  - ProjectConf 
  - Updated conf-variables
  - Preprocessor examples
    - Domain enforcement
    - Maintenance mode
    - VueJS local development
- `app.yaml` serves favicon.png now
- `Index.index` routing for `sitemap.xml`, see https://github.com/viur-framework/viur-core/pull/1157
- Changed `LICENSE` from LGPL-v3 to MIT
- Added missing index for core-3.6 Translation module to `index.yaml`
- Updated `project.json` profile config

Fixes #127.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -4,11 +4,11 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-viur-core = "==3.5.12"
+viur-core = "==3.6.5"
```
Comment (sveneberth) [inline Pipfile:7]:
```suggestion
viur-core = "~=3.6.6"
```

File: Pipfile [hunk 2]
```diff
@@ -4,11 +4,11 @@ verify_ssl = true
...
 
 [dev-packages]
-viur-cli = "==1.1.2"
+viur-cli = "==2.0.1"
```
Comment (sveneberth) [inline Pipfile:10]:
```suggestion
viur-cli = "~=2.0.1"
```

**Review by sveneberth — CHANGES_REQUESTED** [review]
Please update the core

File: project.json [hunk 3]
```diff
@@ -1,29 +1,25 @@
 {
+    "cli-version": "2.0.1",
     "default": {
-        "admin": "4.0.24",
+        "application_name": "{{app_id}}-dev",
```
Comment (sveneberth) [inline project.json:4]:
```suggestion
        "application_name": "{{app_id}}",
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by Grashalmbeisser — APPROVED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**